hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -78,18 +81,17 @@ public void register(Statement statement, boolean cancelable) {
 	public void release(Statement statement) {
 		log.tracev( "Releasing statement [{0}]", statement );
 
-		// Keep this at DEBUG level, rather than warn.  Numerous connection pool implementations can return a
-		// proxy/wrapper around the JDBC Statement, causing excessive logging here.  See HHH-8210.
-		if ( log.isDebugEnabled() && !xref.containsKey( statement ) ) {
-			log.unregisteredStatement();
-		}
-		else {
-			final Set<ResultSet> resultSets = xref.get( statement );
-			if ( resultSets != null ) {
-				closeAll( resultSets );
-			}
+		final Set<ResultSet> resultSets = xref.get( statement );
```

Comment:
Why not just `final Set<ResultSet> resultSets = xref.remove( statement );` here, and no `remove()` call below? Then you would only need to access the map once.

That might change the behavior slightly if `closeAll` throws an exception or somehow uses xref, though. Maybe it's not such a good idea.

Reply:
Yes, you're right, missed that somehow, I think I will revert this change.